assets/src/dashboard/app/views/templates/detail.js

```diff
@@ -99,6 +104,21 @@ function TemplateDetail() {
     });
   }, [templatesOrderById, templates]);
 
+  const relatedTemplates = useMemo(() => {
+    if (templatesOrderById && orderedTemplates) {
+      const getRandomTemplateId = () =>
+        templatesOrderById[
+          Math.floor(Math.random() * templatesOrderById.length)
+        ];
+      const randomlySelectedTemplates = [];
+      for (let i = 0; i < 5; i++) {
+        randomlySelectedTemplates.push(orderedTemplates[getRandomTemplateId()]);
```

Comment:
Also, the UI is getting some duplicate "key" errors in the console, so you might want to update (or even simplify) the random logic to make sure you don't collect duplicate templates.  It could be as a simple as choosing a random start index from 0 to length of templates - 1, and then just counting up 5 templates from that spot (wrapping round when you pass the length).

Reply:
great idea! I changed it to be just a random index - this is great because it allows us to stress test cases where maybe there's only 1 related template.